src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -274,61 +286,92 @@ namespace QgsWms
 
   QByteArray *QgsRenderer::getPrint( const QString &formatString )
   {
-    QStringList layersList, stylesList, layerIdList;
+    //GetPrint request needs a template parameter
+    if ( !mParameters.contains( QStringLiteral( "TEMPLATE" ) ) )
```

Comment:
I think `TEMPLATE` parameter should be managed through QgsWmsParameters  just as the others.